packages/multichain/src/adapters/caip-permission-adapter-eth-accounts.ts

```diff
@@ -0,0 +1,102 @@
+import {
+  type CaipAccountId,
+  type Hex,
+  KnownCaipNamespace,
+  parseCaipAccountId,
+} from '@metamask/utils';
+
+import type { Caip25CaveatValue } from '../caip25Permission';
+import { getUniqueArrayItems, mergeScopes } from '../scope/transform';
+import type { ScopesObject, ScopeString } from '../scope/types';
+import { KnownWalletScopeString, parseScopeString } from '../scope/types';
+
+const isEip155ScopeString = (scopeString: ScopeString) => {
+  const { namespace } = parseScopeString(scopeString);
+
+  return (
+    namespace === KnownCaipNamespace.Eip155 ||
+    scopeString === KnownWalletScopeString.Eip155
+  );
+};
+
+export const getEthAccounts = (
```

Comment:
Nit: Could you add TSDoc entries for each of these exported functions? Ideally we'd have them for all functions, but they're especially useful for the ones that get exported.

Reply:
Added here: https://github.com/MetaMask/core/pull/4784/commits/8194afdbca66c7baf06647d70df5d7add2b3d0dc